### ISSUE TITLE
Sonic Adventure 2 (PAL)

### ISSUE THREAD

**Issue by hatoving** [Rinnegatamante/flycast-vita-compatibility-list#3]
For the record, the game runs okay but it's not very playable due to it stuttering a lot.
There _are_ some spots in the level where it's full-speed, but that's it. 
Disabling Neon SIMD or using Per Triangle Sorting didn't help either (maybe a 30FPS patch for the whole game could help?).
Only minor graphical bugs occurred (i.e. posters clipping in and out of the wall).

Screenshots:
![2022-08-07-154715](https://user-images.githubusercontent.com/64536760/183294392-de46a776-7600-4c08-918d-daf4dd979247.jpg)
![2022-08-07-155332](https://user-images.githubusercontent.com/64536760/183294439-91b30380-c884-4892-a1fa-0a6aaf23d20a.jpg)

Settings used:
Neon SIMD, Simple Shaders and Mipmaps were on.
Auto Frame Skipping was at Maximum.
Downclock Ratio was set to 1.544.

**Comment by Rinnegatamante** [comment]
Ingame+?

**Comment by hatoving** [comment]
Yeah, that fits

**Comment by hatoving** [comment]
Forgot to say, no, self modifying code was set to reduced at it crashes when it's off iirc

**Comment by Rinnegatamante** [comment]
With this patch made by Esppiral, game locks to 30/25 fps and the game feels a lot better. Changing tag to Playabl
[Sonic Adventure 2 v1.008 (2001)(Sega)(NTSC)(US)(M5)[!].gdi.zip](https://github.com/Rinnegatamante/flycast-vita-compatibility-list/files/9387068/Sonic.Adventure.2.v1.008.2001.Sega.NTSC.US.M5.gdi.zip)
e

**Comment by Rii32** [comment]
Thanks Rinne for all the vita love!!! quick question....how do you apply the patch? very new to this so uncertain, hope someone can help

**Comment by Rinnegatamante** [comment]
You have to download the zip and extract the file somehwere (it will contain a cht file). Then in Flycast, you launch the game, press Select, go to Cheats and then press on Load to ux0:. Then you load the said cht file. After that, you'll have two cheats enableable in the Cheats list. Enable both and you're set to go.

**Comment by Brandonheat8** [comment]
For Best performance with the 30fps patch make sure that Float Ops Gamehack is turned off under Advanced Settings, Downclock ratio is set to 1.600 under Advanced Settings, and Broadcast set to NTSC/RGB Component under General Settings.

**Comment by MDashK** [comment]
Detailed config for best performance (so far):

**!!!Make sure to use the 30 FPS patch available above!!!** » The patch works for both PAL and USA NTSC versions of the game.

Dreamcast.Broadcast = 4
Dreamcast.Cable = 3
Dreamcast.Language = 4
Dreamcast.Region = 3
config.Dynarec.downclock = 1.460000
config.Dynarec.smcChecks = 1
config.pvr.AutoSkipFrame = 2
config.rend.ModifierVolumes = yes
config.rend.PerStripSorting = yes
config.rend.WideScreen = yes
config.ta.skip = 0
network.EnableUPnP = no

**All rest should be default values.**
» This applies both to the PAL and NTSC USA version. «
**When using PAL, make sure to select 60Hz instead of 50Hz.**

You can copy-paste the above config directly into the "emu.cfg" file.
Just make sure to specify the Title ID of the game before the first setting (for example, for EUR version is: [MK-5111750] )

Video with example of the performance increase: https://www.youtube.com/watch?v=klFpKkQmaxQ
(Pretty much 30 FPS almost all the level, with a few dips. In the areas that tend to dip, it dips to the 23~24 FPS instead of the 17~18 that used too)

**Comment by Rii32** [comment]
> You have to download the zip and extract the file somehwere (it will contain a cht file). Then in Flycast, you launch the game, press Select, go to Cheats and then press on Load to ux0:. Then you load the said cht file. After that, you'll have two cheats enableable in the Cheats list. Enable both and you're set to go.

Thank you! much appreciated!! absolute GOAT, everyone involved

**Comment by edisionnano** [comment]
I setup flycast for this game but keep getting c2-12828-1 after the second SEGA logo, considering this is marked Playable I believe this shouldn't happen

**Comment by Rinnegatamante** [comment]
> I setup flycast for this game but keep getting c2-12828-1 after the second SEGA logo, considering this is marked Playable I believe this shouldn't happen

LITERALLY 4-5 messages above yours:
![immagine](https://user-images.githubusercontent.com/9152710/187417556-402f84f2-996d-4ebe-b9e5-444bf200ff42.png)

**Comment by edisionnano** [comment]
> > I setup flycast for this game but keep getting c2-12828-1 after the second SEGA logo, considering this is marked Playable I believe this shouldn't happen
> 
> LITERALLY 4-5 messages above yours: ![immagine](https://user-images.githubusercontent.com/9152710/187417556-402f84f2-996d-4ebe-b9e5-444bf200ff42.png)

Yeah that was it, thanks. I believe the index post should be updated with all the info instead of having to look through the entire thread.

**Comment by YauraRtqt** [comment]
doesn't work at all , it redirects me on the dreamcast home menu.

**Comment by MDashK** [comment]
> doesn't work at all , it redirects me on the dreamcast home menu.

So your logic is: "doesn't work for me, so everyone else is lying." ??
If it's kicking you into the BIOS of the Dreamcast, then your GDI or CDI is incorrect.

**Comment by YauraRtqt** [comment]
yeah sry.

**Comment by YauraRtqt** [comment]
Btw my l1 r1 does not work on flycast i tryed multiple times to remap but they are detected , but they work in normal ps vita game or menu.

**Comment by shachar531** [comment]
hey i have a prblome the game load its geting the sonic team and them crash how to fix?

**Comment by Rinnegatamante** [comment]
Use the recommended settings.

**Comment by Trouble233** [comment]
@MDashK can you please write the recommended settings in English. I think I have them correct but it would be a lot easier if you just said yes or no (or whatever value) next to the settings.

**Comment by Glengo37** [comment]
I can't load the cheat. Selecting "load from ux0" goes to a black screen and forces my vita off. When I power it on and open flycast, no cheats are loaded. I've disabled every other plugin aside from the ones that come with the latest modding method (and the newest repatch), no change. Any help?

**Comment by iwantsa2flycastbuticantfindthelink** [comment]
hey @Rinnegatamante. i need a link for the playable sonic adventure 2. i got the cdi, and there was a bug where it would start then crash, can i get a gdi or cdi for the sa2?

**Comment by destined67443** [comment]
> hey [@Rinnegatamante](https://github.com/Rinnegatamante). i need a link for the playable sonic adventure 2. i got the cdi, and there was a bug where it would start then crash, can i get a gdi or cdi for the sa2?

I had the same problem so basically if your rom came with a file that had .cue delete that and everything will be fine
